### PR TITLE
feat(core): Add RSA circuit example

### PR DESCRIPTION
**Pull request by Meyanis95**
Enabling to prove RSA signature.

### CODE REVIEW

File: mopro-core/examples/circom/rsa/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+target
+node_modules
```
Comment (oskarth) [inline mopro-core/examples/circom/rsa/.gitignore:2]: newline

File: mopro-core/examples/circom/rsa/rsa.circom [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    for (var i = 0; i < k; i++) {
+        bigPow.base[i] <== signature[i];
+        bigPow.modulus[i] <== modulus[i];
+    }
+    // By construction of the padding, the padded message is necessarily
+    // smaller than the modulus. Thus, we don't have to check that bigPow is fully reduced.
+    for (var i = 0; i < k; i++) {
+        bigPow.out[i] === padder.padded_message[i];
+    }
+}
```
Comment (oskarth) [inline mopro-core/examples/circom/rsa/rsa.circom:156]: newline

File: mopro-core/src/middleware/circom/mod.rs [hunk 3]
```diff
@@ -510,4 +517,155 @@ mod tests {
...
+            "0",
+            "0",
+            "0",
+            "0",
+            "0",
+            "0",
+            "0",
+          ];
+
+        // let expected_output_vec = vec![0, 0, 0, 0, 0, 0, 0, 0, 32, 65, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 128, 171, 176, 96, 15, 67, 180, 73];
```
Comment (oskarth) [inline mopro-core/src/middleware/circom/mod.rs:641]: I assume there's no specific output that we have to check to ensure validity, is this correct?

**Review by oskarth — APPROVED** [review]
Awesome!

Just minor newline nits and a question, otherwise LGTM

File: mopro-core/src/middleware/circom/mod.rs (hunk 3 above)
Comment (Meyanis95) [reply mopro-core/src/middleware/circom/mod.rs:641]: Yes exactly!
The circuit is verifying the correctness of the RSA signature, but for privacy purposes, it's not outputting anything.

**Comment by oskarth** [comment]
Gotcha!

Would you mind fixing up newline and using conventional commit message? Something like `feat(core): Add RSA circuit example`, then we can merge this!

**Pull request merged**